packages/swap/src/routes/v2/swap.ts

```diff
@@ -196,8 +201,9 @@ router.get(
                     rolledSwaps.lastExecutedChunk && getSwapFields(rolledSwaps.lastExecutedChunk),
                   currentChunk: rolledSwaps.currentChunk && getSwapFields(rolledSwaps.currentChunk),
                   executedChunks: rolledSwaps.executedChunks,
-                  remainingChunks:
-                    (swapDepositChannel?.dcaNumberOfChunks ?? 1) - rolledSwaps.executedChunks,
+                  remainingChunks: isExternal
+                    ? (swapDepositChannel?.dcaNumberOfChunks ?? 1) - rolledSwaps.executedChunks
+                    : (swapRequest?.dcaNumberOfChunks ?? 1) - rolledSwaps.executedChunks,
```

Comment:
we shouldn't use the deposit channel anywhere anymore
```suggestion
                 (swapRequest?.dcaNumberOfChunks ?? 1) - rolledSwaps.executedChunks,
```

Reply:
Changed